File: 725. Split Linked List in Parts.py
class Solution:
    def splitListToParts(self, head: Optional[ListNode], k: int) -> List[Optional[ListNode]]:

        temporrary, length = head, 0
    
        while temporrary: length += 1; temporrary = temporrary.next; 

        result, remnant, response = length // k, length % k, []

        for i in range(k):
            current = head
            previous = head
            for _ in range(result + (1 if i < remnant else 0)):
                try: previous = head; head = head.next; 
                except: break
            try: previous.next = None
            except: pass

            response.append(current)
                
        return response

File: 725. Split Linked List in Parts.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode[] splitListToParts(ListNode head, int k) {
        ListNode arr[]=new ListNode[k];
        int length=0;
        ListNode temp=head;
        while(temp!=null){
            length++;
            temp=temp.next;
        }
        int quo=length/k;
        int rem=length%k;
        temp=head;
        for(int i=0;i<k;i++){
            ListNode partHead=temp;
            ListNode partTail=null;
            int current=quo+(i<rem?1:0);
        for(int j=0;j<current;j++){
            partTail=temp;
            if(temp!=null)
            temp=temp.next;
        }
        if(partTail!=null)
        partTail.next=null;
        arr[i]=partHead;
        }
        return arr;
}
}
